src/demo-app/progress-bar/progress-bar-demo.ts

```diff
@@ -0,0 +1,28 @@
+import {Component} from 'angular2/core';
+import {MdButton} from '../../components/button/button';
+import {MdProgressBar} from '../../components/progress-bar/progress-bar';
+
+@Component({
+    selector: 'progress-bar-demo',
+    templateUrl: 'demo-app/progress-bar/progress-bar-demo.html',
+    styleUrls: ['demo-app/progress-bar/progress-bar-demo.css'],
+    directives: [MdProgressBar, MdButton]
+})
+export class ProgressBarDemo {
+  determinateProgressValue: number = 30;
+  bufferProgressValue: number = 30;
+  bufferBufferValue: number = 40;
+
+  stepDeterminateProgressVal(val: number, ref: string) {
+    this.determinateProgressValue += val;
+  }
+
+  stepBufferProgressVal(val: number) {
+    this.bufferProgressValue += val;
+  }
+
+  stepBufferBufferVal(val: number) {
+    this.bufferBufferValue += val;
+  }
+
```

Comment:
Not in this PR, but in a follow-up could you add a demo that fills up / resets on a timer?